File: hackerrank/project_euler/problem_2/project_euler_2_even_fibonacci_numbers.js
// https://www.hackerrank.com/contests/projecteuler/challenges/euler002/problem

function main() {
    let t = parseInt(readLine());
    for (let a0 = 0; a0 < t; a0++) {
        var n = BigInt(readLine());
        let totalEven = 0n;
        let prev = 1n;
        let curr = 1n;
        while (curr <= n) {
            if (curr % 2n === 0n) {
                totalEven += curr;
            }
            [prev, curr] = [curr, curr + prev];
        }
        console.log(totalEven.toString());
    }
}

File: hackerrank/project_euler/problem_2/project_euler_2_even_fibonacci_numbers.py
# https://www.hackerrank.com/contests/projecteuler/challenges/euler002/problem


def evenFib(n):
    a, b = 1, 2
    sum = 0
    while(a <= n):
        sum += a if a % 2 == 0 else 0
        a, b = b, a+b
    return sum
